Http/Http.cs

```diff
@@ -0,0 +1,352 @@
+	// Http.Instance.Send(UnityWebRequest.Post("uri", "payload"), onNetworkError: () => { });
+	public sealed class Http : MonoBehaviour
+	{
+		private static Http instance;
+
+		public static Http Instance
+		{
+			get
+			{
+				if (instance == null)
+				{
+					instance = new GameObject("Http").AddComponent<Http>();
+				}
+
+				return instance;
+			}
+		}
+
+		private Dictionary<string, string> superHeaders;
+
+		private void Awake()
+		{
+			instance = this;
+			instance.hideFlags = HideFlags.HideInHierarchy;
+
+			superHeaders = new Dictionary<string, string>();
+		}
+
+		/// <summary>
+		/// Super headers are key value pairs that will be added to every subsequent HttpRequest.
+		/// </summary>
+		/// <returns></returns>
+		public Dictionary<string, string> GetSuperHeaders()
+		{
+			return superHeaders;
+		}
+
+		/// <summary>
+		/// Sets a header to the SuperHeaders key value pair, if the header key already exists, the value will be replaced.
+		/// </summary>
+		/// <param name="key">The header key to be set.</param>
+		/// <param name="value">The header value to be assigned.</param>
+		public void SetSuperHeader(string key, string value)
+		{
+			superHeaders[key] = value;
+		}
+
+		/// <summary>
+		/// Removes a header from the SuperHeaders list.
+		/// </summary>
+		/// <param name="key">The header key to be removed.</param>
+		/// <returns></returns>
+		public bool RemoveSuperHeader(string key)
+		{
+			return superHeaders.Remove(key);
+		}
+
+		#region Creation Helpers
+
+		/// <summary>
+		///   <para>Creates a HttpRequest configured for HTTP GET.</para>
+		/// </summary>
+		/// <param name="uri">The URI of the resource to retrieve via HTTP GET.</param>
+		/// <returns>
+		///   <para>A HttpRequest object configured to retrieve data from uri.</para>
+		/// </returns>
+		public static HttpRequest Get(string uri)
+		{
+			return new HttpRequest(UnityWebRequest.Get(uri));
+		}
+
+		/// <summary>
+		///   <para>Create a HttpRequest configured to send form data to a server via HTTP POST.</para>
+		/// </summary>
+		/// <param name="uri">The target URI to which form data will be transmitted.</param>
+		/// <param name="postData">Form body data. Will be URLEncoded via WWWTranscoder.URLEncode prior to transmission.</param>
+		/// <returns>
+		///   <para>A HttpRequest configured to send form data to uri via POST.</para>
+		/// </returns>
+		public static HttpRequest Post(string uri, string postData)
+		{
+			return new HttpRequest(UnityWebRequest.Post(uri, postData));
+		}
+
+		/// <summary>
+		///   <para>Create a HttpRequest configured to send json data to a server via HTTP POST.</para>
+		/// </summary>
+		/// <param name="uri">The target URI to which json data will be transmitted.</param>
+		/// <param name="json">Json body data.</param>
+		/// <returns>
+		///   <para>A HttpRequest configured to send json data to uri via POST.</para>
+		/// </returns>
+		public static HttpRequest PostAsJson(string uri, string json)
```

Comment:
I am still personally not a fan of these being on the HTTP request itself. I think they can be utilities, but the purist in me doesn't like mixing like this.

Reply:
Yes, I did say I was going to move that, I will add it to a utils class

Reply:
+1